cluster-autoscaler/simulator/framework/handle.go

```diff
@@ -68,3 +78,22 @@ func NewHandle(informerFactory informers.SharedInformerFactory, schedConfig *sch
 		DelegatingLister: sharedLister,
 	}, nil
 }
+
+func profileWithDraPlugin(profile *schedulerconfig.KubeSchedulerProfile) *schedulerconfig.KubeSchedulerProfile {
+	result := profile.DeepCopy()
+	addPluginIfNotPresent(result.Plugins.PreFilter, draplugin.Name)
+	addPluginIfNotPresent(result.Plugins.Filter, draplugin.Name)
+	addPluginIfNotPresent(result.Plugins.Reserve, draplugin.Name)
+	return result
+}
```

Comment:
It occurred to me casually "hmm, could we refactor this for more slice-y expressiveness here, or make `addPluginIfNotPresent` accept a slice of `schedulerconfig.PluginSet`s? And then it occurred to me that I wouldn't dare do that without UT. So here's a basic suggestion here:

```suggestion
func profileWithDraPlugin(profile *schedulerconfig.KubeSchedulerProfile) *schedulerconfig.KubeSchedulerProfile {
	result := profile.DeepCopy()
	if result != nil && result.Plugins != nil {
		for _, pluginSet := range []schedulerconfig.PluginSet{
			result.Plugins.PreFilter,
			result.Plugins.Filter,
			result.Plugins.Reserve,
		} {
			addPluginIfNotPresent(pluginSet, draplugin.Name)
		}
	}
	return result
}
```

And here's a quick and dirty `handle_test.go` that will protect the existing implementation (this includes my nil handling additions) so in case someone else thinks to refactor later the UT will ensure that the outcomes don't change in unexpected ways.

```golang
/*
Copyright 2024 The Kubernetes Authors.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

package framework

import (
	"testing"

	"github.com/google/go-cmp/cmp"

	draplugin "k8s.io/kubernetes/pkg/scheduler/framework/plugins/dynamicresources"

	schedulerconfig "k8s.io/kubernetes/pkg/scheduler/apis/config"
	schedulerconfiglatest "k8s.io/kubernetes/pkg/scheduler/apis/config/latest"
)

// TestProfileWithDraPlugin tests that the profileWithDraPlugin function correctly adds the DRA plugin to the profile.
func TestProfileWithDraPlugin(t *testing.T) {
	schedConfig, err := schedulerconfiglatest.Default()
	cmp.Diff(err, nil)
	cmp.Diff(len(schedConfig.Profiles) > 0, true)
	schedProfile := &schedConfig.Profiles[0]
	tests := []struct {
		name               string
		profile            *schedulerconfig.KubeSchedulerProfile
		expectedDRAPlugins bool
	}{
		{
			name:               "* to struct literal input",
			profile:            &schedulerconfig.KubeSchedulerProfile{},
			expectedDRAPlugins: false,
		},
		{
			name:               "nil input",
			profile:            nil,
			expectedDRAPlugins: false,
		},
		{
			name:               "default",
			profile:            schedProfile,
			expectedDRAPlugins: true,
		},
	}

	for _, test := range tests {
		test := test
		t.Run(test.name, func(t *testing.T) {
			actual := profileWithDraPlugin(test.profile)
			found := map[string]bool{
				"PreFilter": false,
				"Filter":    false,
				"Reserve":   false,
			}
			if test.expectedDRAPlugins {
				for _, enabledPreFilterPlugins := range actual.Plugins.PreFilter.Enabled {
					if enabledPreFilterPlugins.Name == draplugin.Name {
						found["PreFilter"] = true
					}
				}
				for _, enabledFilterPlugins := range actual.Plugins.Filter.Enabled {
					if enabledFilterPlugins.Name == draplugin.Name {
						found["Filter"] = true
					}
				}
				for _, enabledReservePlugins := range actual.Plugins.Reserve.Enabled {
					if enabledReservePlugins.Name == draplugin.Name {
						found["Reserve"] = true
					}
				}
				cmp.Diff(found, map[string]bool{
					"PreFilter": true,
					"Filter":    true,
					"Reserve":   true,
				})
			}
		})
	}
}
```

Reply:
Oof, thanks for keeping me honest with the unit testing here. I tried your test and it turns out the implementation doesn't actually add the DRA plugin to the config at all :facepalm: I assumed it does because the integration tests pass, but I checked and the DRA plugin is actually just present in the default config. So no modifications to the plugin sets should be needed here, I removed that part.

I'm pretty sure this was needed before 1.32, but maybe it was a red herring all along..

Reply:
> DRA plugin is actually just present in the default config

this discovery isn't problematic, is it?

Reply:
It shouldn't be, the plugin is only doing anything if the relevant DRA feature gate is enabled in the cluster. This is what I expected to happen all along - we shouldn't need to hack the plugin sets, the plugins should be enabled/disabled via feature gates without CA having to get involved.